cli/sql.go

```diff
@@ -82,15 +104,19 @@ func processOneLine(db *sql.DB, line string) error {
 	// Format all rows using tabwriter.
 	tw := tabwriter.NewWriter(os.Stdout, 0, 8, 0, '\t', 0)
 	fmt.Fprintf(tw, "%s\n", strings.Join(cols, "\t"))
+	sqlVals := make([]sqlValue, len(cols))
 	strs := make([]string, len(cols))
 	vals := make([]interface{}, len(cols))
 	for rows.Next() {
-		for i := range vals {
-			vals[i] = &strs[i]
+		for i := range sqlVals {
+			vals[i] = &sqlVals[i]
 		}
 		if err := rows.Scan(vals...); err != nil {
 			return util.Errorf("scan error: %s", err)
 		}
+		for i := range vals {
+			strs[i] = sqlVals[i].value
+		}
```

Comment:
Using `strings.Join` was convenient when the values where in a `[]string`. Now that you're having to place them in there, it would better to get rid of `strs` and replace the above loop with:

```
  for _, v := range sqlVals {
    fmt.Fprintf(w, "%s\t", v.value)
  }
  fmt.Fprintf(w, "\n")
```